### ISSUE TITLE
Mage can't find template files

### ISSUE THREAD

**Issue by Travior** [mage-ai/mage-ai#1139]
I have this problem whenever I try to create a new block (except for transformers and sensors for some reason), where jinja can't load the template.
The template files are present in `C:\Users\benja\anaconda3\envs\mage-test\Lib\site-packages\mage_ai\data_preparation\templates`
in the folders data_exporters, data_loaders etc... Which I think should be the path where they are loaded from

I'm on Windows 11
Browser is Chrome (however this shouldn't matter)

Error:
```
Traceback (most recent call last):

  File "C:\Users\benja\anaconda3\envs\mage-test\lib\site-packages\tornado\web.py", line 1702, in _execute
    result = method(*self.path_args, **self.path_kwargs)
  File "C:\Users\benja\anaconda3\envs\mage-test\lib\site-packages\mage_ai\server\api\blocks.py", line 96, in post
    upstream_block_uuids=payload.get('upstream_blocks', []),
  File "C:\Users\benja\anaconda3\envs\mage-test\lib\site-packages\mage_ai\data_preparation\models\block\__init__.py", line 338, in create
    pipeline_type=pipeline.type if pipeline is not None else None,
  File "C:\Users\benja\anaconda3\envs\mage-test\lib\site-packages\mage_ai\data_preparation\templates\template.py", line 82, in load_template
    pipeline_type=pipeline_type,
  File "C:\Users\benja\anaconda3\envs\mage-test\lib\site-packages\mage_ai\data_preparation\templates\template.py", line 61, in fetch_template_source
    template_source = __fetch_data_loader_templates(config, pipeline_type=pipeline_type)
  File "C:\Users\benja\anaconda3\envs\mage-test\lib\site-packages\mage_ai\data_preparation\templates\template.py", line 107, in __fetch_data_loader_templates
    template_env.get_template(template_path).render(
  File "C:\Users\benja\anaconda3\envs\mage-test\lib\site-packages\jinja2\environment.py", line 997, in get_template
    return self._load_template(name, globals)
  File "C:\Users\benja\anaconda3\envs\mage-test\lib\site-packages\jinja2\environment.py", line 958, in _load_template
    template = self.loader.load(self, name, self.make_globals(globals))
  File "C:\Users\benja\anaconda3\envs\mage-test\lib\site-packages\jinja2\loaders.py", line 125, in load
    source, filename, uptodate = self.get_source(environment, name)
  File "C:\Users\benja\anaconda3\envs\mage-test\lib\site-packages\jinja2\loaders.py", line 194, in get_source
    pieces = split_template_path(template)
  File "C:\Users\benja\anaconda3\envs\mage-test\lib\site-packages\jinja2\loaders.py", line 35, in split_template_path
    raise TemplateNotFound(template)
jinja2.exceptions.TemplateNotFound: data_loaders\default.jinja
data_loaders\default.jinja
```

**Comment by tommydangerous** [comment]
👀 looking

**Comment by Travior** [comment]
As far as I can tell this appears to be a bug (?) in jinjas template splitting function which seems to be incompatible with windows "\\" directory seperator.
The bug doesn't appear with transformers and sensors since their functions don't use `os.path.join(template_folder, 'default.jinja')`
but rather generate the path with forward slashes, which don't break jinja (e.g., `template_path = 'transformers/default.jinja'`)
I could create an issue in the jinja repo to see if this behavior is intentional, but could also create a PR that fixes the problem regardless of whether this interaction is wanted from jinja's side

**Comment by tommydangerous** [comment]
Wow, thank you so much for figuring out the root cause. 

"I could create an issue in the jinja repo to see if this behavior is intentional, but could also create a PR that fixes the problem regardless of whether this interaction is wanted from jinja's side"

^ +1 to that. This would be amazing, thank you so much! 🙏

**Comment by Travior** [comment]
So I've done some digging in the jinja issues and found this thread https://github.com/pallets/jinja/issues/1378#issuecomment-812410922

To me, it seems like they intend this behavior, meaning that paths that use the `split_template_path` function should probably be sanitized (maybe just with a simple `path.replace("\\", "/")`?)

**Comment by tommydangerous** [comment]
@lucafaggianelli we’re working on this fix, so sorry for the inconvenience.

**Comment by wangxiaoyou1993** [comment]
@Travior @lucafaggianelli This bug has been fixed in version 0.7.19. Let us know if you still have template loading issue in windows.

**Comment by wangxiaoyou1993** [comment]
PR: https://github.com/mage-ai/mage-ai/pull/1476
